### PR TITLE
Add `to_local_time()` in function reference docs

### PR DESCRIPTION
**Pull request by appletreeisyellow**
## Which issue does this PR close?

Follow-up PR for https://github.com/apache/datafusion/pull/11347#issuecomment-2220519822

Relates to https://github.com/apache/datafusion/issues/10602, https://github.com/apache/datafusion/issues/11358

## Rationale for this change

A scalar UDF `to_local_time()` was added in https://github.com/apache/datafusion/pull/11347. This PR documents `to_local_time()` in the scalar function reference page. 

## What changes are included in this PR?

After adding the document for `to_local_time()`, the doc page will look like this:

Screenshot 2024-07-10 at 2 51 54 PM

## Are these changes tested?

No. Only adding documents, so it shouldn't break anything.

## Are there any user-facing changes?

No

### CODE REVIEW

File: docs/source/user-guide/sql/scalar_functions.md [hunk 1]
```diff
@@ -1771,6 +1772,68 @@ to_date(expression[, ..., format_n])
...
+```
+> SELECT to_local_time('2024-04-01T00:00:20Z'::timestamp);
++---------------------------------------------+
+| to_local_time(Utf8("2024-04-01T00:00:20Z")) |
++---------------------------------------------+
+| 2024-04-01T00:00:20                         |
++---------------------------------------------+
+
+> SELECT to_local_time('2024-04-01T00:00:20Z'::timestamp AT TIME ZONE 'Europe/Brussels');
+SELECT to_local_time('2024-04-01T00:00:20Z'::timestamp AT TIME ZONE 'Europe/Brussels');
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:1799]: This line looks duplicated -- we can probably remove it
```suggestion
```

File: docs/source/user-guide/sql/scalar_functions.md [hunk 2]
```diff
@@ -1771,6 +1772,68 @@ to_date(expression[, ..., format_n])
...
+FROM (
+  SELECT '2024-04-01T00:00:20Z'::timestamp AT TIME ZONE 'Europe/Brussels' AS time
+);
++---------------------------+------------------------------------------------+---------------------+-----------------------------+
+| time                      | type                                           | to_local_time       | to_local_time_type          |
++---------------------------+------------------------------------------------+---------------------+-----------------------------+
+| 2024-04-01T00:00:20+02:00 | Timestamp(Nanosecond, Some("Europe/Brussels")) | 2024-04-01T00:00:20 | Timestamp(Nanosecond, None) |
++---------------------------+------------------------------------------------+---------------------+-----------------------------+
+
+# combine `to_local_time()` with `date_bin()`
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:1820]:
```suggestion
# combine `to_local_time()` with `date_bin()` to bin on boundaries in the timezone rather
# than UTC boundaries
```

**Review by alamb — APPROVED** [review]
Looks great -- thank you @appletreeisyellow

**Comment by alamb** [comment]
cc @Abdullahsab3

File: docs/source/user-guide/sql/scalar_functions.md (hunk 1 above)
Comment (appletreeisyellow) [reply docs/source/user-guide/sql/scalar_functions.md]: Thanks for catching that! Removed in https://github.com/apache/datafusion/pull/11401/commits/61c005f5532243fc4bf68c01b33b289168e8a130

File: docs/source/user-guide/sql/scalar_functions.md (hunk 2 above)
Comment (appletreeisyellow) [reply docs/source/user-guide/sql/scalar_functions.md]: Updated in https://github.com/apache/datafusion/pull/11401/commits/61c005f5532243fc4bf68c01b33b289168e8a130

**Comment by alamb** [comment]
Thanks again @appletreeisyellow !

**Pull request merged**